File: sample/use_cocoapods/objectivec/objectivec/GoogleReward/GoogleRewardViewController.m
#import "GoogleRewardViewController.h"
@import GoogleMobileAds;

@interface GoogleRewardViewController () <GADFullScreenContentDelegate>

@property (weak, nonatomic) IBOutlet UITextField *unitIdView;
@property (weak, nonatomic) IBOutlet UIButton *buttonLoad;
@property (weak, nonatomic) IBOutlet UIButton *buttonShow;

@property(nonatomic, strong) GADRewardedAd *rewardedAd;

@end

@implementation GoogleRewardViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    _buttonShow.enabled = NO;
}

- (IBAction)downButtonLoad:(id)sender {
    NSString* unitId = _unitIdView.text;
    if (unitId.length == 0) {
        return;
    }

    GAMRequest *request = [GAMRequest request];
    [GADRewardedAd loadWithAdUnitID:unitId request:request completionHandler:^(GADRewardedAd *ad, NSError *error) {
        if (error) {
            self.buttonLoad.enabled = YES;
            NSLog(@"ViewController: downButtonLoad error = %@", error.localizedDescription);
            return;
        }
        NSLog(@"ViewController: downButtonLoad ad loaded.");
        self.buttonShow.enabled = YES;
        self.rewardedAd = ad;
        self.rewardedAd.fullScreenContentDelegate = self;
    }];

    _buttonLoad.enabled = NO;
}

- (IBAction)downButtonShow:(id)sender {
    if (self.rewardedAd) {
        NSLog(@"ViewController: downButtonShow show.");
        [self.rewardedAd presentFromRootViewController:self userDidEarnRewardHandler:^ {
            GADAdReward *reward = self.rewardedAd.adReward;
            NSLog(@"ViewController: rewardedAd:userDidEarnRewardHandler type = %@, amount = %lf", reward.type, [reward.amount doubleValue]);
        }];
    } else {
        NSLog(@"ViewController: downButtonShow not show.");
    }
    self.buttonLoad.enabled = YES;
}


#pragma mark: GADFullScreenContentDelegate

/// Tells the delegate that the rewarded ad was presented.
- (void)adDidPresentFullScreenContent:(id)ad {
    NSLog(@"ViewController: adDidPresentFullScreenContent");
}

/// Tells the delegate that the rewarded ad failed to present.
- (void)ad:(id)ad
    didFailToPresentFullScreenContentWithError:(NSError *)error {
    NSLog(@"ViewController: didFailToPresentFullScreenContentWithError error = %@", [error localizedDescription]);
}

/// Tells the delegate that the rewarded ad was dismissed.
- (void)adDidDismissFullScreenContent:(id)ad {
    NSLog(@"ViewController: adDidDismissFullScreenContent");
    _buttonShow.enabled = NO;
    _rewardedAd = nil;
}


@end

File: sample/use_cocoapods/swift/swift/GoogleReward/GoogleRewardViewController.swift
import UIKit
import GoogleMobileAds

class GoogleRewardViewController: UIViewController {

    var unitId: String!
    var rewardedAd: GADRewardedAd!

    @IBOutlet weak var unitIdView: UITextField!
    @IBOutlet weak var buttonLoad: UIButton!
    @IBOutlet weak var buttonShow: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        buttonShow.isEnabled = false
    }

    @IBAction func downButtonLoad(_ sender: Any) {
        let unitId = unitIdView.text
        if (unitId!.isEmpty) {
            return
        }

        let request = GAMRequest()
        GADRewardedAd.load(withAdUnitID: unitId!, request: request, completionHandler: { (ad, error) in
            if let error = error {
                self.buttonLoad.isEnabled = true
                print("ViewController: downButtonLoad error = \(error.localizedDescription)")
                return
            }
            print("ViewController: downButtonLoad ad loaded.")
            self.buttonShow.isEnabled = true
            self.rewardedAd = ad
            self.rewardedAd?.fullScreenContentDelegate = self
        })

        buttonLoad.isEnabled = false
    }
    
    @IBAction func downButtonShow(_ sender: Any) {
        if let ad = rewardedAd {
            print("ViewController: downButtonShow show.")
            ad.present(fromRootViewController: self, userDidEarnRewardHandler: {
                let reward = ad.adReward
                print("ViewController: rewardedAd:userDidEarnRewardHandler type = \(reward.type), amount = \(reward.amount).")
            })
        } else {
            print("ViewController: downButtonShow not show.")
        }
        self.buttonLoad.isEnabled = true
    }
    
}
// MARK: GADFullScreenContentDelegate implementation
extension GoogleRewardViewController : GADFullScreenContentDelegate {
    /// Tells the delegate that the rewarded ad was presented.
    func adWillPresentFullScreenContent(_ ad: GADFullScreenPresentingAd) {
        print("ViewController: adWillPresentFullScreenContent")
    }

    /// Tells the delegate that the rewarded ad was dismissed.
    func adDidDismissFullScreenContent(_ ad: GADFullScreenPresentingAd) {
        print("ViewController: adDidDismissFullScreenContent")
        buttonShow.isEnabled = false
        rewardedAd = nil
    }

    /// Tells the delegate that the rewarded ad failed to present.
    func ad(_ ad: GADFullScreenPresentingAd, didFailToPresentFullScreenContentWithError error: Error) {
        print("ViewController: didFailToPresentFullScreenContentWithError error = \(error.localizedDescription)")
    }

}

